File: 0-tarefas/JavaApplication_OrdenaAlunos/src/Aluno.java

/**
 *
 * @author alexandrezamberlan
 */
public class Aluno implements Comparable<Aluno> {
    public int matricula;
    public String nome;
    public String curso;

    public Aluno(int matricula, String nome, String curso) {
        this.matricula = matricula;
        this.nome = nome;
        this.curso = curso;
    }
    
    @Override
    public int compareTo(Aluno aluno) {
        if (this.matricula < aluno.matricula) {
            return -1;
        }
        return 1;
    }

    
    @Override
    public boolean equals(Object o) {
        if (o instanceof Aluno) {
            Aluno a = (Aluno)o;
            return this.matricula == a.matricula;
        }
        return false;
    }
}

File: 13-talvezDiscutir/09-colecoes/ColecaoCSharp/Aluno.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ColecaoCSharp
{
    public class Aluno : IComparable<Aluno>
    {
        public int Id { get; set; }
        public string Nome { get; set; }

        public Aluno(int id, string nome)
        {
            Id = id;
            Nome = nome;
        }

        public override boolean Equals(Object o) {
            if ((o == null) || !this.GetType().Equals(obj.GetType())) {
                return false;
            } else {
                Aluno e = (Aluno) o;
                return (Id == e.Id && this.Nome == e.Nome);
            }
        }
        

        public int CompareTo(Aluno? other)
        {
            if (this.Id < other.Id)
                return -1;
            if (this.Id == other.Id)
            {
                if (this.Nome.CompareTo(other.Nome) < 0)
                    return -1;
            }
            return 1;
        }        
    }
}
